File: qqbot-module-platform-admin/src/main/java/com/mieai/qqbot/app/web/SpaForwardController.java
package com.mieai.qqbot.app.web;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

/** Keeps Angular client-side routes working when the application serves the UI directly. */
@Controller
public class SpaForwardController {

    @GetMapping({"/", "/login", "/setup"})
    public String forwardAdminRoutes() {
        return "forward:/index.html";
    }

    @GetMapping("/modules/**")
    public String forwardModuleRoutes() {
        return "forward:/index.html";
    }
}

File: qqbot-module-platform-admin/src/main/kotlin/com/mieai/qqbot/app/web/SpaForwardController.kt
package com.mieai.qqbot.app.web

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping

/** Keeps Angular client-side routes working when the application serves the UI directly. */
@Controller
class SpaForwardController {
    @GetMapping(value = ["/", "/login", "/setup"])
    fun forwardAdminRoutes(): String = "forward:/index.html"

    @GetMapping("/modules/**")
    fun forwardModuleRoutes(): String = "forward:/index.html"
}
